packages/dds/tree/src/util/utils.ts

```diff
@@ -484,3 +484,17 @@ export function capitalize<S extends string>(s: S): Capitalize<S> {
 export function compareStrings<T extends string>(a: T, b: T): number {
 	return a > b ? 1 : a === b ? 0 : -1;
 }
+
+/**
+ * Returns a function which, when invoked for the first time, will run `evaluate` and cache its result for future invocations.
+ * @remarks Use the `evaluated` property of the returned function to check if the value has been cached yet.
+ */
+export function lazy<T>(evaluate: () => T): (() => T) & { readonly evaluated: boolean } {
```

Comment:
`core-utils` has a lazy type that functions like this. Should we just use that?

Reply:
What is it called? I know we have some stuff for promises but I haven't used them before.

Reply:
`Lazy`

Reply:
Thanks, good suggestion, I've used it.